Disable garbage collection during compile_time_instructions count in benchmark base by default.

laithsakka opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #135768

When we measure compile time instruction count, probably we do want in most cases to measure gc instructions
disabling it here by default. 
if it is needed we can add an option to allow it, or someone can use the regular total instruction count instead of compile time instruction count.
cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @rec

laithsakka committed:
disabel gc during compile_time_instructions count in benchmark base

[ghstack-poisoned]

ezyang commented on benchmarks/dynamo/pr_time_benchmarks/benchmark_base.py:
```diff
@@ -117,6 +120,7 @@ def _count_compile_time_instructions(self):
             results.append(count)
 
         config.record_compile_time_instruction_count = False
+        gc.enable()
```
You need to try...finally this in case of exception?

ezyang commented:
It's definitely right, from a stability perspective, to disable GC. I do want to punish code that excessively uses lots of memory, since this can increase GC time. But maybe this is not enough of an impact to measure.

laithsakka commented:
> It's definitely right, from a stability perspective, to disable GC. I do want to punish code that excessively uses lots of memory, since this can increase GC time. But maybe this is not enough of an impact to measure.

Maybe for that a metric that tracks total allocations or peak memory https://docs.python.org/3/library/tracemalloc.html

laithsakka committed:
Update on "Disable garbage collection during compile_time_instructions count in benchmark base by default."


When we measure compile time instruction count, probably we do want in most cases to measure gc instructions
disabling it here by default. 
if it is needed we can add an option to allow it, or someone can use the regular total instruction count instead of compile time instruction count.
cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

laithsakka committed:
Update on "Disable garbage collection during compile_time_instructions count in benchmark base by default."


When we measure compile time instruction count, probably we do want in most cases to measure gc instructions
disabling it here by default. 
if it is needed we can add an option to allow it, or someone can use the regular total instruction count instead of compile time instruction count.
cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

laithsakka commented on benchmarks/dynamo/pr_time_benchmarks/benchmarks/basic_modules_benchmarks.py:
```diff
@@ -131,7 +131,8 @@ def name(self):
     def _prepare_once(self):
         self.m = self.ModuleClass()
-        self.input = torch.ones(10)
+        torch.set_float32_matmul_precision("high")
```
ah rebase trash

laithsakka committed:
Update on "Disable garbage collection during compile_time_instructions count in benchmark base by default."


When we measure compile time instruction count, probably we do want in most cases to measure gc instructions
disabling it here by default. 
if it is needed we can add an option to allow it, or someone can use the regular total instruction count instead of compile time instruction count.
cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

laithsakka committed:
Update on "Disable garbage collection during compile_time_instructions count in benchmark base by default."


When we measure compile time instruction count, probably we do want in most cases to measure gc instructions
disabling it here by default. 
if it is needed we can add an option to allow it, or someone can use the regular total instruction count instead of compile time instruction count.
cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

laithsakka commented:
@pytorchbot merge

laithsakka committed:
Update on "Disable garbage collection during compile_time_instructions count in benchmark base by default."


When we measure compile time instruction count, probably we do want in most cases to measure gc instructions
disabling it here by default. 
if it is needed we can add an option to allow it, or someone can use the regular total instruction count instead of compile time instruction count.
cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx chenyang78 kadeng chauhang amjames rec

[ghstack-poisoned]

laithsakka commented:
@pytorchbot merge
